### DISCUSSION TITLE
biblatex types vs overall list of references [pdf output]

### DISCUSSION THREAD

**Discussion by giabaio** [quarto-dev/quarto-cli#5528] [Q&A]
I have a simple pdf document where I want to create a list of references, from a `bib` file. That's easy enough but I'd like to make full use of `biblatex` and be able to divide the publications by type. 
 
For example, I can do
```
---
title: "List of publications"
author: Gianluca Baio
date: today
bibliography: mypubs.bib
csl: harvard.csl
nocite: |
   @*
format: 
   pdf:
     cite-method: biblatex
     biblatexoptions: maxbibnames=99, sorting=ydnt, style=authoryear, dashed=false,
        terseinits=true, date=year
---

\printbibliography[type=book,title={Books}]
\printbibliography[type=article,title={Journal articles}]
```

This works great (assumes of course that `mypubs.bib` and `harvard.csl` are available...). BUT, in addition to printing the two separate lists of books and articles (as required by the two `\printbibliography` commands), it also automatically generates at the end of the resulting pdf document a full list of references (by effect of the `nocite: @*` command), which however I don't want --- that's the point of having it divided by types.

I've tried adding a `suppress-bibliography: true` argument to the `yaml`, but to no effect.

Am I missing something obvious or is there an extra option hidden somewhere that I should use (say a `biblatex-type` or something argument to the `yaml`?)

Thanks
Gianluca

**Comment by mcanouil** [comment]
If you don't want the default template or part of it, customise it using partials, see <https://quarto.org/docs/journals/templates.html#template-partials>.

Additionally, take a look at the following Quarto extension: <https://github.com/pandoc-ext/section-bibliographies>.

**Reply by giabaio** [reply]
Thanks, that'll probably do! Let me have a look --- if I find something useful, I can follow up on this with a revised template!

**Comment by giabaio** [comment]
OK. Thanks --- I think it's fairly simple, but I have another question, if that's OK? It may also be trivial, but not sure, again, I'm missing an obvious problem...

Say the `qmd` file looks like
```
---
title: "List of publications"
author: Gianluca Baio
date: today
bibliography: mypubs.bib
csl: harvard.csl
nocite: |
   @*
mainfont: "Arial"
include-in-header: 
   text: |
    \newfontfamily\sectionfont[Color=ff7832]{Arial}
    \newfontfamily\subsectionfont[Color=MediumBlue]{Arial}
    \newfontfamily\subsubsectionfont[Color=Green]{Arial}
    \addtokomafont{section}{\sectionfont}
    \addtokomafont{subsection}{\subsectionfont}
    \addtokomafont{subsubsection}{\subsubsectionfont}
format: 
   pdf:
     cite-method: biblatex
     biblatexoptions: maxbibnames=99, sorting=ydnt, style=authoryear, dashed=false,
        terseinits=true, date=year
     show-bib: false
---

\printbibliography[type=book,title={Books}]
```
(**NB**: that's pretty much like I showed before, with some extra customisation of the fonts & colours in the sections as per [here](https://stackoverflow.com/questions/73662721/specify-custom-fonts-and-attributes-for-h1-h6-headings-in-quarto-pdf)). All works OK (except the full reference list is also included by the default partials).

Now, I can modify the `biblio.tex` partial as
```
$if(biblatex)$
$if(show-biblio)$
$if(beamer)$
\begin{frame}[allowframebreaks]{$biblio-title$}
  \bibliographytrue
  \printbibliography[heading=none]
\end{frame}
$else$
\printbibliography$if(biblio-title)$[title=$biblio-title$]$endif$
$endif$
$endif$
$endif$
```
(basically I now include an extra `if` depending on the value of  `show-biblio`, which only prints the full reference list if that's set to `true` --- this is basically enough for what I wanted to do).

However, in this case, the `include-in-header` commands seem to clash with the rest of the file and the pdf doesn't get compiled with an error
```
compilation failed- error
Undefined control sequence.
<recently read> \addtokomafont 
                               
l.83 \addtokomafont
                   {section}{\sectionfont} 

see publication-list.log for more information.
```

I don't really understand why --- does this mean I would need to add other partials, even if all I want to modify is a simple bit in the `biblio.tex` file? (I know there are probably more elegant ways to do this -- I've not worked with partials before, really, so just trying to get a grasp on that...).

Thanks & sorry if I'm being thick!

**Reply by mcanouil** [reply]
Don't use the include option, set the corresponding partials to avoid the clash.
Look at the whole template and if you add LaTeX code (in include) that conflicts with the template, then you have no other choice than modifying a partial or the whole template.

**Reply by giabaio** [reply]
OK. Thanks. I think the solution is in this case to add specifically the option
```
   pdf:
     documentclass: scrartcl
```
to the yaml, otherwise the resulting `tex` file thinks that the class is `article` and things are broken.

Everything else now works in the `include-in-header` part (which doesn't clash anymore).

Hope this may be helpful for people.

**Comment by cderv** [comment]
> I've tried adding a suppress-bibliography: true argument to the yaml, but to no effect.

This will only works with Pandoc's citation processing, but here you are using `biblatex` by specifying the `cite-method`

Just wanted to give a quick explainer around this :)

**Comment by giabaio** [comment]
Understood! Thanks!
